### PR TITLE
Svelte: welcome banner followups

### PR DESCRIPTION
**Pull request by camdencheek**
A few followups from #64163. Comments inline.

## Test plan

Manual local testing

### CODE REVIEW

File: client/web-sveltekit/src/routes/+layout.svelte [hunk 1]
```diff
@@ -111,10 +112,7 @@
     <slot />
 </main>
 
-<WelcomeOverlay
-    show={(process.env.PW_TEST === 'true' && !$welcomeOverlayDismissed) ?? false}
```
Comment (camdencheek) [inline client/web-sveltekit/src/routes/+layout.svelte:115]: This `PW_TEST` thing wasn't actually doing what I thought it was, and caused the banner to not be shown at all when deployed.

File: client/web-sveltekit/src/routes/WelcomeOverlay.svelte [hunk 2]
```diff
@@ -83,7 +83,7 @@
                 background: var(--color-background, #f9fafb);
             }
             :global(.theme-dark) & {
-                background: var(--color-background, #f9fafb);
+                background: var(--color-background, #0f111a);
```
Comment (camdencheek) [inline client/web-sveltekit/src/routes/WelcomeOverlay.svelte:86]: Used the wrong color for dark mode.

File: client/web-sveltekit/src/testing/integration.ts [hunk 3]
```diff
@@ -117,6 +117,9 @@ class Sourcegraph {
                     },
                 })
             })
+            await this.page.addInitScript(() =>
+                window.localStorage.setItem('temporarySettings', '{"webNext.welcomeOverlay.dismissed": true}')
+            )
```
Comment (camdencheek) [inline client/web-sveltekit/src/testing/integration.ts:122]: In playwright tests, don't ever show the overlay

File: client/web/src/sveltekit/SvelteKitNavItem.tsx [hunk 4]
```diff
@@ -40,7 +40,8 @@ export const SvelteKitNavItem: FC<{ userID?: string }> = ({ userID }) => {
         return null
     }
 
-    const showDeparture = !departureDismissed
+    // only show if the welcome message has been dismissed so we know they have been introduced to the new webapp
+    const showDeparture = !departureDismissed && welcomeDismissed
```
Comment (camdencheek) [inline client/web/src/sveltekit/SvelteKitNavItem.tsx:44]: When a user is not in a default-on environment, we do not want to show the departure message if they have never activated the rewrite

**Review by vovakulikov — APPROVED** [review]

**Pull request merged**